data_structures/queue/priority_queue.py

```diff
@@ -0,0 +1,241 @@
+"""
+Pure Python implementation of Priority Queue using lists
+"""
+
+
+class FixedPriorityQueue:
+    """
+    In a Priority Queue the elements are entred as an when the come
+    But while removing or deleting an element the highest priority element is deleted
+    in FIFO fashion.
+    Here the lowest integer has the highest priority.
+    Example:
+    priority(0) > priority(5)
+    priority(16) > priority(32)
+    Here as an example I have taken only 3 priorities viz. 0, 1, 2
+    0 Being the highest priority and 2 being the lowest
+    You can change the priorities as per your need
+
+    Examples
+    >>> FPQ = FixedPriorityQueue()
+    >>> FPQ.enqueue(0, 10)
+    >>> FPQ.enqueue(1, 70)
+    >>> FPQ.enqueue(0, 100)
+    >>> FPQ.enqueue(2, 1)
+    >>> FPQ.enqueue(2, 5)
+    >>> FPQ.enqueue(1, 7)
+    >>> FPQ.enqueue(2, 4)
+    >>> FPQ.enqueue(1, 64)
+    >>> FPQ.enqueue(0, 128)
+    >>> FPQ.print_queue()
+    Priority 0: [10, 100, 128]
+    Priority 1: [70, 7, 64]
+    Priority 2: [1, 5, 4]
+    >>> FPQ.dequeue()
+    10
+    >>> FPQ.dequeue()
+    100
+    >>> FPQ.dequeue()
+    128
+    >>> FPQ.dequeue()
+    70
+    >>> FPQ.dequeue()
+    7
+    >>> FPQ.print_queue()
+    Priority 0: []
+    Priority 1: [64]
+    Priority 2: [1, 5, 4]
+    >>> FPQ.dequeue()
+    64
+    >>> FPQ.dequeue()
+    1
+    >>> FPQ.dequeue()
+    5
+    >>> FPQ.dequeue()
+    4
+    >>> print(FPQ.dequeue())
+    Traceback (most recent call last):
+        ...
+    """
+
+    def __init__(self):
+        self.queue = [
+            [],
+            [],
+            [],
+        ]
+
+    def enqueue(self, priority, data):
+        """
+        This function enters the element into the FixedPriorityQueue based on its priority
+        If the priority is invalid an Exception is raised saying Invalid Priority!
+        If the FixedPriorityQueue is full an Exception is raised saying Over Flow!
+        """
+        if priority > 2:
+            raise Exception("Invalid Priority!")
+        elif len(self.queue[priority]) == 100:
+            raise Exception("Over Flow!")
+        else:
+            self.queue[priority].append(data)
+
+    def dequeue(self):
+        """
+        This function returns the highest priority element from the FixedPriorityQueue
+        in FIFO fashion
+        If the FixedPriorityQueue is empty and this function is called then an 
+        exception is raised saying Under Flow!
+        """
+        if not self.queue[0] and not self.queue[1] and not self.queue[2]:
+            raise Exception("Under Flow!")
+
+        if len(self.queue[0]) != 0:
+            return self.queue[0].pop(0)
+        elif len(self.queue[1]) != 0:
+            return self.queue[1].pop(0)
+        else:
+            return self.queue[2].pop(0)
+
+    def print_queue(self):
```

Comment:
Instead, please do [`def __str__(self) -> str:`](https://docs.python.org/3/reference/datamodel.html?highlight=__str__#object.__str__) and then we can use `str(my_queue)` or `print(my_queue)` to see the results.

Reply:
How do you do that?

Reply:
Replace `"Your code goes here..."` with the str that you want to see.